x-pack/plugins/alerting/server/alerts_client/alerts_client.ts

```diff
@@ -242,9 +242,15 @@ export class AlertsClient<
       this.reportedAlerts[alert.id] = alert.payload;
     }
 
+    const alertDoc: AlertData | undefined =
+      !alert.payload && !!this.fetchedAlerts.data[alert.id]
+        ? (this.fetchedAlerts.data[alert.id] as AlertData)
```

Comment:
I remove the cast and my IDE did not complain. Do you think we need it with the generic you specified above?